components/camera/client.go

```diff
@@ -625,6 +628,23 @@ func (c *client) Unsubscribe(ctx context.Context, id rtppassthrough.Subscription
 	return nil
 }
 
+func (c *client) trackName() string {
+	// if c.conn is a *grpc.SharedConn then the client
+	// is talking to a module and we need to send the fully qualified name
+	if _, ok := c.conn.(*grpc.SharedConn); ok {
+		return c.Name().String()
+	}
+	// otherwise we know we are talking to either a main or remote robot part
+
+	trackName := c.Name().SDPTrackName()
+	if c.remoteName != "" {
```

Comment:
@dgottlieb I know you asked me to make this switch based on the connection type rather than whether or not the resource graph passed the component's client a remoteName, however I can't do that as both in the case of `client -> main part` and `main part -> remote part` the conn type will be `ReconfigurableClientConn`.

Reply:
This seems fine. I thought the predicate wasn't a simple "empty or not" check. And if it's not empty, I can see the code is predictably popping off one remote name.

And I also see the remote name is explicitly passed into the constructor. Seems perfectly fine to depend on long-term.